include/phasar/Utils/LLVMShorthands.h

```diff
@@ -216,6 +217,23 @@ bool isVarAnnotationIntrinsic(const llvm::Function *F);
  *
  */
 llvm::StringRef getVarAnnotationIntrinsicName(const llvm::CallInst *CallInst);
+
+class ModulesToSlotTracker {
+  friend class ProjectIRDB;
+  friend class LLVMBasedICFG;
+  friend class LLVMZeroValue;
+
+private:
+  static inline llvm::SmallDenseMap<const llvm::Module *,
+                                    std::unique_ptr<llvm::ModuleSlotTracker>, 2>
+      MToST;
+
+  static void updateMSTForModule(const llvm::Module *);
+  static void deleteMSTForModule(const llvm::Module *);
```

Comment:
What's the idea behind this design? On the one side, these functions are private, on the other there are three friend classes that access the "private" functionality.

Reply:
We don't want anyone to mess with the slot trackers, but also we don't want to introduce new inheritances. And also we want to keep llvmIRTo*String as free functions.

Reply:
kk